logging/src/main/java/life/qbic/logging/impl/publisher/PublisherFactory.java

```diff
@@ -0,0 +1,26 @@
+package life.qbic.logging.impl.publisher;
+
+import life.qbic.logging.api.Publisher;
+import life.qbic.logging.impl.subscriber.error.EmailOnErrorSubscriber;
+
+/**
+ * Factory for creating logging publishers providing the {@link Publisher} interface.
+ *
+ * @since 1.0.0
+ */
+public class PublisherFactory {
+
+  /**
+   * Creates a simple publisher that includes a {@link EmailOnErrorSubscriber} subscription.
+   *
+   * @return an instance of type {@link Publisher}
+   * @since 1.0.0
+   */
+  public static Publisher createSimplePublisher() {
+    var publisher = new SimplePublisher();
+    var emailOnErrorSubscriber = new EmailOnErrorSubscriber();
+    publisher.subscribe(emailOnErrorSubscriber);
```

Comment:
I would not expect the returned publisher to have a subscriber already attached to it. Can you motivate your thoughts here a bit?

Reply:
Sure :) We need to encapsulate our subscriber policies somewhere and not give the responsibility to the application. I imagine to have some default subscriber like `EmailOnError` and `SlackMessageOnError` that are hooked in all the time and the application can provide implementations (still need to define the interface for these) or if not, provide no-op implementations. With this we can decide within the logging component, that we want a slack message subscriber executed for all errors, write an email for warnings to our mailing list or whatever. We can decide on application level to provide an implementation for sending emails or slack messages or default to no-op.
But the rules are defined in the component, that is where i think it belongs. Therefore I preconfigure the publisher in the factory :)

Reply:
Thanks for the elaborate motivation :+1:
I it makes sense to have a default subscription strategy in place.